### ISSUE TITLE
[FEATURE] maybe we should add some decomposition to the generated code?

### ISSUE THREAD

**Issue by MaxmaxmaximusAWS** [pmndrs/gltfjsx#46]
maybe we should add some decomposition to the generated code?

for example, in addition to the default export, also export each model and texture separately, by name.

**Comment by tapjay** [comment]
This would be an incredible feature, especially when using with typescript.

**Comment by drcmda** [comment]
sounds interesting. so you mean something like this?

```jsx
export function Arms() {
  const { nodes, material } = useGLTF("model.glb")
  return ...
}

export function Legs() {
  const { nodes, material } = useGLTF("model.glb")
  return ...
}

export default function Model() {
  return (
    <group>
      <Arms />
      <Legs />
```

just not sure about the specifics and if it's doable. lots of questions come to mind. for instance, names can easily break javascript conventions.

**Comment by tapjay** [comment]
Yes exactly. I can imagine there are lots of edge cases around doing this properly but it would be great to be able to run an exported scene from blender through gltfjsx and then selectively import objects into the scene (and get type errors if something has been changed).

**Comment by drcmda** [comment]
you can edit the main component though and make its parts conditional, isn't that enough? either way, you can make a pr and then we can review it. there is zero chance that i will have the time for that right now.

**Comment by tapjay** [comment]
@drcmda I'm doing that for now, if I find I encounter wanting to do this regularly I'll make a PR. Thanks for your work on this.

**Comment by MaxmaxmaximusAWS** [comment]
> so you mean something like this?

exactly =)

**Comment by drcmda** [comment]
one more concern would be how to treat n levels of nesting. if it's all about exposing "nodes" then each would just contain the mesh with the linked up geometry and material, but groups and their positioning would be lost in the named export. but yeah, make a flag and wire it into the code - i would probably accept it.
